utils/riff/index.js

```diff
@@ -212,3 +212,23 @@ export const PeerColors = [
     Colors.darkfern,
     Colors.aquamarine,
 ];
+
+/**
+ * returns a truncated string, with a maximum character count of maxLength
+ * if no maxLength passed, 100 is used by default
+ * an ending to tag onto the end of the string can be passed, by default an ellipsis(...) is applied
+ */
+export function textTruncate(string, maxLength, ending) {
```

Comment:
This function has several issues stylistically.
1. It uses the javascript type 'string' as a parameter name
1. double equal comparison instead of triple (I realize that when comparing to null it supports undefined as well as null, but it's not obvious, I had to look that up). My take is that null is an odd value to pass in and it might as well be an error, so compare === undefined, or better yet use default values if that's what you really want.
1. don't modify parameters, use new local variables

```
/**
 * Limit the given string to the specifed number of characters.
 * If the string has more characters than the max allowed
 * a new string will be returned which is exactly max characters long
 * consisting of a suffix of an ellipsis character (U+2026) OR the given
 * missing character string, and the 1st characters from the given
 * string to make the returned sting have the max length.
 *
 * @param {string} s - The string whose length should not exceed maxLen
 * @param {number} maxLen - The maximum length of the returned string. default: 100
 * @param {string} missingCharSuffix - The suffix to replace the truncated
 *      end of the returned string with. default: ellipsis character
 *
 * @returns a string with a maximum of maxLen characters
 */
export function textTruncate(s, maxLen = 100, missingCharSuffix = '\u2026') {
    if (s.length <= maxLen) {
        return s;
    }

    return s.slice(0, maxLen - missingCharSuffix.length) + missingCharSuffix;
}
```

Reply:
Oh I started to fix these things, but I am assuming that you're comment is the function that you would prefer...I have replaced the original...I did forget about the ability to use default values for parameters in javascript..nice touch!

Reply:
I'm not seeing the updated textTruncate function in the PR

Reply:
Sorry about that, it is in a new commit.

Reply:
Just keeping you honest! :smile:

Reply:
ha well that definitely needed to be updated, thanks.